addons/mail/controllers/main.py

```diff
@@ -154,20 +160,26 @@ def read_followers(self, res_model, res_id):
         followers = []
         follower_id = None
         for follower in follower_recs:
-            if follower.partner_id == request.env.user.partner_id:
-                follower_id = follower.id
-            followers.append({
+            follower_data = {
                 'id': follower.id,
                 'partner_id': follower.partner_id.id,
                 'channel_id': follower.channel_id.id,
                 'name': follower.name,
                 'display_name': follower.display_name,
                 'email': follower.email,
                 'is_active': follower.is_active,
-                # When editing the followers, the "pencil" icon that leads to the edition of subtypes
-                # should be always be displayed and not only when "debug" mode is activated.
-                'is_editable': True
-            })
+            }
+            if follower.partner_id == request.env.user.partner_id:
+                follower_id = follower.id
+                follower_data['is_editable'] = True
+            else:
+                try:
+                    request.env[res_model].check_access_rights("write")
+                    request.env[res_model].browse(res_id).check_access_rule("write")
```

Comment:
Maybe it is cached, but somehow if we have 30K followers (e.g. elearning) are you really going to do the same calls 30K times ?